src/mission_start.cpp

```diff
@@ -55,528 +55,558 @@ point random_house_in_closest_city()
  * reveal the area around it (uses overmapbuffer::reveal with reveal_rad),
  * and returns the mission target.
  */
-point target_om_ter(const std::string &omter, int reveal_rad, mission *miss, bool must_see)
+tripoint target_om_ter( const std::string &omter, int reveal_rad, mission *miss, bool must_see )
 {
     int dist = 0;
-    const point place = overmap_buffer.find_closest(
-        g->u.global_omt_location(), omter, dist, must_see);
-    if(place != overmap::invalid_point && reveal_rad >= 0) {
-        overmap_buffer.reveal(place, reveal_rad, g->get_levz());
+    const tripoint place = overmap_buffer.find_closest(
+                            g->u.global_omt_location(), omter, dist, must_see );
+    if( place != overmap::invalid_tripoint && reveal_rad >= 0 ) {
+        overmap_buffer.reveal( place, reveal_rad );
     }
     miss->set_target( place );
     return place;
 }
 
-point target_om_ter_random(const std::string &omter, int reveal_rad, mission *miss, bool must_see)
+tripoint target_om_ter_random( const std::string &omter, int reveal_rad, mission *miss,
+                               bool must_see )
 {
     int dist = 0;
-    std::vector<point> places = overmap_buffer.find_all(
-        g->u.global_omt_location(), omter, dist, must_see);
-    if (places.size() == 0){
-        debugmsg("Couldn't find %s", omter.c_str());
-        return point();
+    auto places = overmap_buffer.find_all( g->u.global_omt_location(), omter, dist, must_see );
+    if( places.size() == 0 ) {
+        debugmsg( "Couldn't find %s", omter.c_str() );
+        return overmap::invalid_tripoint;
     }
     const auto &cur_om = g->get_cur_om();
-    std::vector<point> places_om;
-    for (auto &i : places) {
-        if (&cur_om == overmap_buffer.get_existing_om_global(i))
-            places_om.push_back(i);
+    std::vector<tripoint> places_om;
+    for( auto &i : places ) {
+        if( &cur_om == overmap_buffer.get_existing_om_global( i ) ) {
+            places_om.push_back( i );
+        }
     }
-    const point place = places_om[rng(0,places_om.size()-1)];
-    if(place != overmap::invalid_point && reveal_rad >= 0) {
-        overmap_buffer.reveal(place, reveal_rad, g->get_levz());
+
+    const tripoint place = places_om[rng( 0, places_om.size() - 1 )];
+    if( place != overmap::invalid_tripoint && reveal_rad >= 0 ) {
```

Comment:
You can remove the check `place != overmap::invalid_tripoint`, `place` is from the vector returned by `find_all` and should never by an invalid point.